mteb/tasks/Reranking/multilingual/MIRACLReranking.py

```diff
@@ -1,38 +1,277 @@
+        """Computes the metrices in a batched way, by batching all queries and
+        all documents together
+        """
+        # using encode_queries and encode_corpus functions if they exists,
+        # which can be defined by users to add different instructions for query and passage conveniently
+        encode_queries_func = (
+            model.encode_queries
+            if isinstance(model, EncoderWithQueryCorpusEncode)
+            else model.encode
+        )
+        encode_corpus_func = (
+            model.encode_corpus
+            if isinstance(model, EncoderWithQueryCorpusEncode)
+            else model.encode
+        )
+
+        logger.info("Encoding queries...")
+        if isinstance(self.samples[0]["query"], str):
+            all_query_embs = np.asarray(
+                encode_queries_func(
+                    [sample["query"] for sample in self.samples],
+                    batch_size=self.batch_size,
+                )
+            )
+        elif isinstance(self.samples[0]["query"], list):
+            # In case the query is a list of strings, we get the most similar embedding to any of the queries
+            all_query_flattened = [
+                q for sample in self.samples for q in sample["query"]
+            ]
+            all_query_embs = np.asarray(
+                encode_queries_func(all_query_flattened, batch_size=self.batch_size)
+            )
+        else:
+            raise ValueError(
+                f"Query must be a string or a list of strings but is {type(self.samples[0]['query'])}"
+            )
+
+        logger.info("Encoding candidates...")
+        all_docs = []
+        for sample in self.samples:
+            all_docs.extend(sample["candidates"])
+
+        all_docs_embs = np.asarray(
+            encode_corpus_func(all_docs, batch_size=self.batch_size)
+        )
+
+        # Compute scores
+        logger.info("Evaluating...")
+        query_idx, docs_idx = 0, 0
+        results, qrels = {}, {}
+        for instance in self.samples:
+            num_subqueries = (
+                len(instance["query"]) if isinstance(instance["query"], list) else 1
+            )
+            query_emb = all_query_embs[query_idx : query_idx + num_subqueries]
+            query_idx += num_subqueries
+
+            positive = instance["positive"]
+            docs = instance["candidates"]
+            num_doc = len(docs)
+            docs_emb = all_docs_embs[docs_idx : docs_idx + num_doc]
+            docs_idx += num_doc
+
+            fake_qid = str(query_idx)
+            results[fake_qid] = self.rerank(query_emb, docs_emb)
+            qrels[fake_qid] = {
+                str(i): 1 if doc in positive else 0 for i, doc in enumerate(docs)
+            }
+
+        ndcg, _map, recall, precision = RetrievalEvaluator.evaluate(
+            qrels=qrels, results=results, k_values=self.k_values
+        )
+        return {**ndcg, **_map, **recall, **precision}
```

Comment:
I just saw this. Thanks for taking care of the remaining issues! apologized that I was distracted by other work.

Here it needs to append the "(MIRACL)" after metrics as well?